cachers/disk.go

```diff
@@ -61,30 +60,28 @@ func (dc *DiskCache) Get(ctx context.Context, actionID string) (outputID, diskPa
 		// Protect against malicious non-hex OutputID on disk
 		return "", "", nil
 	}
-	output2 := ie.OutputID[:2]
-	return ie.OutputID, filepath.Join(dc.Dir, output2, fmt.Sprintf("o-%v", ie.OutputID)), nil
+	return ie.OutputID, dc.OutputFilename(ie.OutputID), nil
 }
 
 func (dc *DiskCache) OutputFilename(outputID string) string {
-	if len(outputID) < 4 || len(outputID) > 1000 {
+	if !validHex(outputID) {
 		return ""
 	}
-	for i := range outputID {
-		b := outputID[i]
-		if b >= '0' && b <= '9' || b >= 'a' && b <= 'f' {
-			continue
-		}
+	return filepath.Join(dc.Dir, outputID[:2], fmt.Sprintf("o-%s", outputID))
+}
+
+func (dc *DiskCache) ActionFilename(actionID string) string {
+	if !validHex(actionID) {
 		return ""
 	}
-	return filepath.Join(dc.Dir, fmt.Sprintf("o-%s", outputID))
```

Comment:
The diff is rendering in a confusing way because it looks like this is for action ID now, but the bug was that this returned path was missing the component `outputID[:2]` in the middle.